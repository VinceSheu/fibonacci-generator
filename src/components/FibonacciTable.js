import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';


function FibonacciTable(props) {
  const { fibonacciArray } = props;

  return (
    <table className="fib-table">
      <thead>
        <tr>
          <th className="left-column">n</th>
          <th>F(n)</th>
        </tr>
      </thead>
      <tbody>
      {
        fibonacciArray.map( (fibonacciNumber, index) => (
          <tr> 
            <td className="left-column">{index}</td>
            <td>{fibonacciNumber}</td>
          </tr>
        ))
      }
      </tbody>
    </table>
  );
}

FibonacciTable.propTypes = {
  fibonacciArray: PropTypes.array.isRequired,
};

export default FibonacciTable;