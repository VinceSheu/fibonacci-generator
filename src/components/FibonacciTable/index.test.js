import React from 'react';
import { shallow } from 'enzyme';

import FibonacciTable from './index';

describe('<FibonacciTable />', () => {
  it('should render FibonacciTable correctly', () => {
    const renderedComponent = shallow(<FibonacciTable fibonacciArray={[0,1,1,2,3,5]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});