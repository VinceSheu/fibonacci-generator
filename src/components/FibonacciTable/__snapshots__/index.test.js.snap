// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FibonacciTable /> should render FibonacciTable correctly 1`] = `
<table
  className="fib-table"
>
  <thead>
    <tr>
      <th
        className="left-column"
      >
        n
      </th>
      <th>
        F(n)
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td
        className="left-column"
      >
        0
      </td>
      <td>
        0
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        className="left-column"
      >
        1
      </td>
      <td>
        1
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        className="left-column"
      >
        2
      </td>
      <td>
        1
      </td>
    </tr>
    <tr
      key="2"
    >
      <td
        className="left-column"
      >
        3
      </td>
      <td>
        2
      </td>
    </tr>
    <tr
      key="3"
    >
      <td
        className="left-column"
      >
        4
      </td>
      <td>
        3
      </td>
    </tr>
    <tr
      key="5"
    >
      <td
        className="left-column"
      >
        5
      </td>
      <td>
        5
      </td>
    </tr>
  </tbody>
</table>
`;
